packages/data/src/components/use-select/index.js

```diff
@@ -104,7 +105,100 @@ const renderQueue = createQueue();
  * ```
  * @return {UseSelectReturn<T>} A custom react hook.
  */
+function Selecter( registry ) {
+	const NOTHING = {};
+	let lastMapSelect = NOTHING;
+	let lastMapResult = NOTHING;
+	let lastMapResultValid = false;
+
+	const NULLSCRIBER = () => {
+		lastMapResultValid = false;
+		return () => {};
+	};
+	let subscribe = NULLSCRIBER;
+
+	function createSubscriber( stores ) {
+		if ( stores.length === 0 ) {
+			return NULLSCRIBER;
+		}
+
+		return ( lis ) => {
+			lastMapResultValid = false;
+			const unsubs = stores.map( ( storeName ) => {
+				return registry.subscribe( () => {
+					lastMapResultValid = false;
+					lis();
+				}, storeName );
+			} );
+
+			return () => {
+				// The return value of the subscribe function could be undefined if the store is a custom generic store.
+				unsubs.forEach( ( unsub ) => unsub?.() );
+			};
+		};
+	}
+
+	function select( mapSelect, resub ) {
```

Comment:
This will break for all instances that access `useSelect()` with this syntax, won't it:

```
import { store } from '@wordpress/core-data';
...
useSelect( store );
...
```

I believe we should support that syntax if we use the `useSelect` API. That means we'll need to add the workaround that exists in the old `useSelect()` implementation.

Reply:
Yes, this part is still missing from the new version. Also, it turns out we don't have any unit tests for it!